### DISCUSSION TITLE
pnpm in rush

### DISCUSSION THREAD

**Discussion by foora** [pnpm/pnpm#3961] [Q&A] [answered]
* - "global" - use PNPM's global store, which has the benefit of being shared
     *    across multiple repo folders, but the disadvantage of less isolation for builds
     *    (e.g. bugs or incompatibilities when two repos use different releases of PNPM)

I saw this in `rush.json` which is the config file [rushstack](https://github.com/microsoft/rushstack) .
`bugs or incompatibilities when two repos use different releases of PNPM` ,does PNPM really have this problem?

**Comment by zkochan** [comment, marked as answer]
No, I don't think there are known issues with the pnpm store. IMHO, the shared content-addressable store is one of the best features of pnpm. It doesn't make sense not to use it.

**Comment by chengcyber** [comment]
@foora In my mind, it's irrelavant with pnpm itself. Mostly, it comes from Rush.js feature, declaring a specified pnpm version used in current monorepo by setup `pnpmVersion` property in `rush.json`. 

In this way, there are chances you setup different pnpmVersion in two repos accessing same "global" pnpm store, which potentionally cause issues. such as using pnpm@4 globally, while using pnpm@6 in monorepo.

So, if you want to set `pnpmStore` to `global`. You should make sure you(and your team members) install a pnpm without breaking change in pnpm-store. For instance, install pnpm@latest in global, declare `pnpmVersion` to `6.0.0` in monorepo A, and declare `pnpmVersion` to `6.10.0` in monorepo B. Both A and B are managed by Rush.js.

**Reply by zkochan** [reply]
but we version the store. So if there are breaking changes to the store, pnpm creates a different subdirectory for it.

**Reply by zkochan** [reply]
The current store version is v3

![image](https://user-images.githubusercontent.com/1927579/140772002-9962f563-3aac-40cc-938b-550190535ab4.png)

if let's say pnpm v7 will have breaking changes to the store, it will write to a v4 directory

**Reply by chengcyber** [reply]
Or, is there a chance that repo A uses npm official registry while repo B uses company private registry, leading to feature request on the isolation of pnpm store?
